_posts/2020-12-02-solidity-underhanded-contest-winners.md

```diff
@@ -0,0 +1,100 @@
+
+Now, let's have a look at this year's winners, starting from the 5th place!
+
+## Solidity Underhanded Contest 2020 Winners
+
+### 5️⃣ th Place: [Marius van der Wijden](https://github.com/ethereum/solidity-underhanded-contest/tree/master/submissions_2020/submission14) 
+
+_commentary by [Alex Beregszaszi](https://twitter.com/alexberegszaszi)_
+
+This submission highlights the need for clearly distinguishing error conditions from return values. The `ecrecover` precompile has a single output, the address, which is set to zero in case of errors. Many high-level wrappers for this precompile introduce new error conditions, but still reuse the same mechanic of returning a “zero address” on error. This was the case with OpenZeppelin’s version [prior](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/v2.5.0/contracts/cryptography/ECDSA.sol) to the 3.0 series.
+
+Using this validation shortcoming, the submission allows anyone to create unlimited "vote accepted" transactions.
+
+Solidity has always supported multiple return values which allows returning a status/error code and data separately, but more importantly the established good practice is to use `require` statements for validation, which most `ecrecover` use cases follow today.
+
+While the submission is written clearly and the issue is not obvious without dissection both the `closeVote` and the *trusted* `ECDSA.recover` functions, in our opinion there are two shortcomings raising suspicion:
+1. Append only data structures are not commonly used and there have been various published bugs about clearing them.
+2. Awareness about `ECDSA.recover` was raised recently, and thus one would expect a more thorough review of its applications. The submission used the version from OpenZeppelin 2.5.0.
+
+### 4️⃣ th Place: [Richard Moore](https://github.com/ethereum/solidity-underhanded-contest/tree/master/submissions_2020/submission8)
+
+_commentary by [Stefan Beyer](https://twitter.com/beyer_st)_
+
+The submission exploits `CREATE2` to hijack a multisig contract that controls upgradability. Whilst this is a known issue, the scenario demonstrates the impact of `CREATE2` in combination with self-destruct and shows the importance of assessing the contracts deployment procedure in a security analysis. 
+
+### 3️⃣ rd Place: [Cory Dickson](https://github.com/ethereum/solidity-underhanded-contest/tree/master/submissions_2020/submission1)
+
+_commentary by [Goncalo Sá](https://twitter.com/GNSPS)_
+ 
+This submission highlights the perils of validating non-EOA (Externally Owned Accounts) addresses within the EVM. Even though assessing whether an account interacting with a certain system is not a contract has been the source of many production hacks on mainnet and therefore is a well-studied, bad-practice pattern, the one where an user DoS’ a system by self-destructing their contract is not.
```

Comment:
```suggestion
This submission highlights the perils of validating non-EOA (Externally Owned Accounts) addresses within the EVM. Even though assessing whether an account interacting with a certain system is not a contract has been the source of many production hacks on mainnet and therefore is a well-studied, bad-practice pattern, the one where a user DoS’s a system by self-destructing their contract is not.
```
(DoS-es?)

Reply:
Hmm interesting :D My guess would be ``DoS-es``(as a verb, right)?